Unfortunately the error message is a bit weird in case of a compile time assert: 
```
home/sperry/workspace/2.3/src/mixxxapplication.cpp: In constructor ‘MixxxApplication::MixxxApplication(int&, char**)’:
/home/sperry/workspace/2.3/src/mixxxapplication.cpp:79:28:   in ‘constexpr’ expansion of ‘{anonymous}::makei(0)’
/home/sperry/workspace/2.3/src/util/assert.h:57:27: error: call to non-‘constexpr’ function ‘void mixxx_debug_assert(const char*, const char*, int, const char*)’
   57 |         mixxx_debug_assert(#cond, __FILE__, __LINE__, ASSERT_FUNCTION); \
      |         ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sperry/workspace/2.3/src/mixxxapplication.cpp:60:5: note: in expansion of macro ‘DEBUG_ASSERT’
   60 |     DEBUG_ASSERT(i);
      |     ^~~~~~~~~~~~
```
that is also the case with the std assert(), not our fault

--- a/src/util/assert.h
+++ b/src/util/assert.h
@@ -11,7 +11,7 @@ inline void mixxx_debug_assert(const char* assertion, const char* file, int line
 #endif
 
 #ifdef MIXXX_DEBUG_ASSERTIONS_ENABLED
-inline bool mixxx_maybe_debug_assert_return_true(const char* assertion,
+inline bool mixxx_debug_assert_return_true(const char* assertion,
         const char* file,
         int line,
         const char* function) {
@@ -65,7 +65,7 @@ inline void mixxx_release_assert(const char* assertion, const char* file, int li
 #ifdef MIXXX_DEBUG_ASSERTIONS_ENABLED
 #define VERIFY_OR_DEBUG_ASSERT(cond)            \
     if (Q_UNLIKELY(!static_cast<bool>(cond)) && \
-            mixxx_maybe_debug_assert_return_true(#cond, __FILE__, __LINE__, ASSERT_FUNCTION))
+            mixxx_debug_assert_return_true(#cond, __FILE__, __LINE__, ASSERT_FUNCTION))
 #else
 #define VERIFY_OR_DEBUG_ASSERT(cond) if (Q_UNLIKELY(!static_cast<bool>(cond)))
 #endif

